lib/scanner/walk.go

```diff
@@ -80,18 +80,14 @@ type CurrentFiler interface {
 	CurrentFile(name string) (protocol.FileInfo, bool)
 }
 
-type Lstater interface {
-	Lstat(name string) (os.FileInfo, error)
-}
-
 func Walk(cfg Config) (chan protocol.FileInfo, error) {
 	w := walker{cfg}
 
 	if w.CurrentFiler == nil {
 		w.CurrentFiler = noCurrentFiler{}
 	}
-	if w.Lstater == nil {
-		w.Lstater = defaultLstater{}
+	if w.Filesystem == nil {
+		w.Filesystem = fs.DefaultFilesystem
```

Comment:
Not by the tests. It seems sane to have a fallback, but in the end I don't feel strongly about it either way.